src/diart/models.py

```diff
@@ -1,28 +1,41 @@
 from abc import ABC, abstractmethod
-from typing import Optional, Text, Union, Callable
+from typing import Optional, Text, Union, Callable, Mapping, TYPE_CHECKING
 
 import torch
 import torch.nn as nn
+from requests import HTTPError
 
 try:
     import pyannote.audio.pipelines.utils as pyannote_loader
+    from pyannote.audio import Inference, Model
+    from pyannote.audio.pipelines.speaker_verification import (
+        WeSpeakerPretrainedSpeakerEmbedding,
+    )
 
     _has_pyannote = True
 except ImportError:
     _has_pyannote = False
 
+if TYPE_CHECKING:
+    from pyannote.audio.pipelines.speaker_verification import (
+        WeSpeakerPretrainedSpeakerEmbedding,
+    )
+
 
 class PyannoteLoader:
     def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__()
         self.model_info = model_info
         self.hf_token = hf_token
 
-    def __call__(self) -> nn.Module:
-        return pyannote_loader.get_model(self.model_info, self.hf_token)
+    def __call__(self) -> Union[nn.Module, WeSpeakerPretrainedSpeakerEmbedding]:
+        try:
+            return pyannote_loader.get_model(self.model_info)
+        except HTTPError:
+            return WeSpeakerPretrainedSpeakerEmbedding(self.model_info)
 
 
-class LazyModel(nn.Module, ABC):
+class LazyModel(ABC):
     def __init__(self, loader: Callable[[], nn.Module]):
```

Comment:
We need to change the type of the loader here to `Callable[[], Callable]`

Reply:
Done!